### PR TITLE
account for resident role

### PR DESCRIPTION
**Pull request by yosefserkez**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: TT6-186384928

# A brief description of the changes

Current behavior:
The notice script breaks when there is no consumer_role present for example on coverall enrollments with resident_role
New behavior:
The notice script handles both consumer roles and resident roles properly
# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

File: app/domain/operations/notices/ivl_enr_notice_trigger.rb [hunk 1]
```diff
@@ -58,8 +58,7 @@ def build_addresses(person, is_primary)
       end
 
       def update_and_build_verification_types(person)
-        date = TimeKeeper.date_of_record
```
Comment (4till2) [inline app/domain/operations/notices/ivl_enr_notice_trigger.rb:61]: Removal of unused code

File: app/domain/operations/notices/ivl_enr_notice_trigger.rb [hunk 2]
```diff
@@ -83,11 +84,11 @@ def build_family_member_hash(enrollment)
               person_health: { is_tobacco_user: person.is_tobacco_user },
               is_active: person.is_active,
               is_disabled: person.is_disabled,
-              consumer_role: build_consumer_role(person.consumer_role),
               addresses: build_addresses(person, fm.is_primary_applicant)
             }
           }
           member_hash[:person].merge!(verification_types: update_and_build_verification_types(person)) if outstanding_verification_types.present?
+          member_hash[:person].merge!(consumer_role_ref || resident_role_ref)
```
Comment (4till2) [inline app/domain/operations/notices/ivl_enr_notice_trigger.rb:91]: There should only ever be either consumer or resident role

File: app/domain/operations/notices/ivl_enr_notice_trigger.rb [hunk 3]
```diff
@@ -182,7 +184,23 @@ def build_consumer_role(consumer_role)
         }
       end
 
+      def build_resident_role(resident_role)
```
Comment (4till2) [inline app/domain/operations/notices/ivl_enr_notice_trigger.rb:187]: These fields are copied from the related contract

File: app/models/resident_role.rb [hunk 4]
```diff
@@ -181,4 +181,14 @@ def residency_denied?
   def residency_verified?
     is_state_resident?
   end
+
```
Comment (4till2) [inline app/models/resident_role.rb:184]: Copy of the methods in consumer_role.rb

**Review by ATBull81 — DISMISSED** [review]

File: app/domain/operations/notices/ivl_enr_notice_trigger.rb (hunk 2 above)
Comment (mdkaraman) [reply app/domain/operations/notices/ivl_enr_notice_trigger.rb:91]: Confirmed this is the case, but we need to use these methods as our source of truth for which role is active:
```
 def is_consumer_role_active?
    self.active_individual_market_role == "consumer" ? true : false
  end

  def is_resident_role_active?
     self.active_individual_market_role == "resident" ? true : false
  end
```

File: app/domain/operations/notices/ivl_enr_notice_trigger.rb [hunk 5]
```diff
@@ -121,12 +123,11 @@ def build_enrollments_hash(enrollments)
             hbx_enrollment_members: enrollment_member_hash(enr),
             product_reference: product_reference(product, issuer),
             issuer_profile_reference: issuer_profile_reference(issuer),
-            consumer_role_reference: consumer_role_reference(consumer_role),
             is_receiving_assistance: (enr.applied_aptc_amount > 0 || (product.is_csr? ? true : false)),
             timestamp: timestamp(enr)
           }
           enrollment_hash.merge!(special_enrollment_period_reference: special_enrollment_period_reference(enr)) if enr.is_special_enrollment?
-          enrollment_hash
+          enrollment_hash.merge!(consumer_role_ref || resident_role_ref)
```
Comment (mdkaraman) [inline app/domain/operations/notices/ivl_enr_notice_trigger.rb:130]: We should add all roles that present on the enrollment, i.e., add both consumer role and resident role if both exist on the enrollment.

**Review by ATBull81 — APPROVED** [review]

**Review by mdkaraman — APPROVED** [review]
Thank you for leaving the comments on the PR!  Very helpful when reviewing.

Looks good!

**Pull request merged**